ra_aid/agents/ciayn_agent.py

```
@@ -132,19 +202,34 @@ def _build_prompt(self, last_result: Optional[str] = None) -> str:
+        globals_dict = {tool.func.__name__: tool.func for tool in self.tools}
+
         try:
+            code = code.strip()
+            # code = code.replace("\n", " ")
+
+            # if the eval fails, try to extract it via a model call
+            if validate_function_call_pattern(code):
+                functions_list = "\n\n".join(self.available_functions)
+                code = _extract_tool_call(code, functions_list)
+
             result = eval(code.strip(), globals_dict)
```

Ruff (0.8.2)

Use of possibly insecure function; consider using `ast.literal_eval`

(S307)